connectors/src/connectors/slack/bot.ts

```diff
@@ -130,27 +158,101 @@ export async function botAnswerMessageWithErrorHandling(
         connectorId: connector.id,
         slackTeamId,
       },
-      `Unexpected exception answering to Slack Chat Bot message`
+      "Unexpected exception updating mention on Chat Bot message"
     );
     const slackClient = await getSlackClient(connector.id);
     await slackClient.chat.postMessage({
       channel: slackChannel,
-      text: `An unexpected error occured. Our team has been notified.`,
+      text: "An unexpected error occured. Our team has been notified.",
       thread_ts: slackMessageTs,
     });
 
-    return new Err(new Error(`An unexpected error occured`));
+    return new Err(new Error("An unexpected error occured"));
+  }
+}
+
+async function processResult(
+  res: Result<AgentMessageSuccessEvent | undefined, Error>,
+  params: BotAnswerParams,
+  connector: ConnectorResource
+) {
+  if (res.isErr()) {
+    const { slackChannel, slackMessageTs } = params;
+    logger.error(
+      {
+        error: res.error,
+        errorMessage: res.error.message,
+        ...params,
+      },
+      "Failed answering to Slack Chat Bot message"
+    );
+    let errorMessage: string | undefined = undefined;
+    if (res.error instanceof SlackExternalUserError) {
+      errorMessage = res.error.message;
+    } else {
+      errorMessage = `An error occured : ${res.error.message}. Our team has been notified and will work on it as soon as possible.`;
+    }
+
+    const { slackChatBotMessage, mainMessage } =
+      res.error instanceof SlackMessageError
+        ? res.error
+        : { mainMessage: undefined, slackChatBotMessage: undefined };
+
+    const conversationUrl = slackChatBotMessage?.conversationId
```

Comment:
IIRC we already have some of this logic in `stream_conversation_handler` can we consolidate everything in one place?

Reply:
you mean having : 
```
makeDustAppUrl(`/w/${connector.workspaceId}/assistant/${conversation.sId}`)
``` 
in a function ?

Reply:
Yep, just want to make sure that the day we decided to finally move this back to `conversations` instead of `assistant` we don't have hundreds of places to check.